src/bridge/bridge.hpp

```diff
@@ -27,19 +27,79 @@
 #pragma once
 
 #define BRIDGE_USE_PIPES 1
+
 #include "ProtobufMessages.pb.h"
 #include <variant>
 #include <optional>
-#include "../VRDriver.hpp"
+#include <thread>
+#include <mutex>
+#include <queue>
 
 enum BridgeStatus {
     BRIDGE_DISCONNECTED = 0,
     BRIDGE_CONNECTED = 1,
     BRIDGE_ERROR = 2
 };
 
-BridgeStatus runBridgeFrame(SlimeVRDriver::VRDriver &driver);
+namespace SlimeVRDriver {
+    class VRDriver;
+
+    class Bridge {
+    public:
+        enum class EState {
+            BRIDGE_DISCONNECTED = 0,
+            BRIDGE_CONNECTED = 1,
+            BRIDGE_ERROR = 2
+        };
+
+        explicit Bridge(VRDriver *i_pDriver);
```

Comment:
why a pointer over a reference?

Reply:
Because I pass the class pointer from VRDriver -> src/VRDriver.cpp line 9

Reply:
You can still use a reference if you deref the pointer before passing it, like the old code was doing when it was passing a reference to the bridge every time it did something.